autoload/highlightedyank.vim

```diff
@@ -85,6 +85,55 @@ function! s:yank_normal(count, register) abort "{{{
   endif
 endfunction
 "}}}
+function! highlightedyank#autocmd_highlight() abort "{{{
+  let view = winsaveview()
+  let region = {}
+  let region.head = getpos("'[")
+  let region.tail = getpos("']")
+  let char = getline(region.tail[1])[region.tail[2] - 1]
+  let linecontent = v:event.regcontents[len(v:event.regcontents) - 1]
+  let tailchar = linecontent[len(linecontent) - 1]
+  if char != tailchar
+    let region.tail[2] = region.tail[2] - 1
+  endif
```

Comment:
I noticed that, whenever I typed `yw`, for example, it would highlight one extra char than it was supposed to do.

Suppose the following snipet, with the cursor in the first caracter.

```
foo bar baz
```

If I typed `yw`, it was highlighting

```
[foo b]ar baz
```

When it was supposed to highlight

```
[foo ]bar baz
```

In essence, the `']` register seems to behave differently in the TextYankPost when compared to the operatorfunc, in which the position seems to be correcly handled.

Therefore, I used the regcontent variable in order check if the last character yanked corresponds to the `getpos("']")` position. If it doesn't match, it adjusts the position off by one character.

Ain't pretty, but it works. Do you have a suggestion on a better way of doing this?